### PR TITLE
Fix handling of miniscreen when user takes/gives control

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/<ticket-number>) |

#### Main changes
-

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: pt_miniscreen/app.py [hunk 1]
```diff
@@ -64,6 +64,10 @@ def dim(self):
 
     def create_button_handler(self, func):
         def handler():
+            if self.user_has_control:
```
Comment (olivierwilkinson) [inline pt_miniscreen/app.py:67]: Nice, good catch!

File: pt_miniscreen/app.py [hunk 2]
```diff
@@ -96,11 +100,23 @@ def handle_down_button_release(self):
         self.root.scroll_down()
 
     def display(self):
+        def restore_miniscreen():
```
Comment (olivierwilkinson) [inline pt_miniscreen/app.py:103]: This should be put on the class to avoid recreating it on every display, which can be called very frequently.

**Review by olivierwilkinson — COMMENTED** [review]
The code looks good, only need to move the `restore_miniscreen` definition to the class I think.

I am still finding that the app doesn't come back up after running a further challenge so maybe there is something wrong on my device. When you come back online maybe let's go through it and see what the problem is together.
